tests/test_integration.py

```diff
@@ -2,73 +2,33 @@
 from dotenv import find_dotenv, load_dotenv
 import todo_app.app as app
 from unittest.mock import patch, Mock
-
-@pytest.fixture
+import mongomock
+import pymongo
+       
+@pytest.fixture 
 def client():
-    # Use our test integration config instead of the 'real' version
-    
-    file_path = find_dotenv('.env.test')
+    file_path = find_dotenv('.env.test') 
     load_dotenv(file_path, override=True)
+    with mongomock.patch(servers=(('fakemongo.com', 27017),)): 
+        test_app = app.create_app()
+        with test_app.test_client() as client:
+            setup_testdata()
+            yield client
 
-    # Create the new app.
-    test_app = app.create_app()
-
-    # Use the app to create a test_client that can be used in our tests.
-    with test_app.test_client() as client: 
-        yield client
-
-@patch('requests.get')
-def test_index_page(mock_get_requests, client):
+def setup_testdata():
+    db_connection = os.getenv('CLIENT')
+    db_name = os.getenv('DBNAME')
+    connection = pymongo.MongoClient(db_connection)
+    db = connection[db_name]
+    card = {"name": "test1", "status": "To Do"}
+    collection = db['items']
+    collection.insert_one(card)
+    
+def test_index_page(client):
     # Replace call to requests.get(url) with our own function
-    mock_get_requests.side_effect = mock_get_lists
     response = client.get('/')
     assert response.status_code == 200
```

Comment:
Would be nice to check the test data, since you've set it up e.g.

```python
    assert "test1" in str(response.data)
```

Reply:
Done!